satellite/admin-back-office-v1.md

```diff
@@ -0,0 +1,617 @@
+Accountable: TBD
+Consulted:
+- Product. Reference person [Andi Steinke](https://github.com/stoweandi)
+- Customer support team. Reference person [Helene Unland](https://github.com/heunland).
+- Production owners.
+- Team integrations console squad: [Maximillian von Briesen (Moby)](https://github.com/mobyvb).
+Informed:
+- QA team.
+
+### Context
+
+Satellite currently has an [administration
+API](https://github.com/storj/storj/blob/main/satellite/admin/README.md)
+and minimalist [Web interface](https://github.com/storj/storj/tree/main/satellite/admin/ui) which
+allows to interact with its endpoints through Web forms.
+
+API authentication is conducted by two mechanism:
+- [An unique token secret](https://github.com/storj/storj/blob/main/satellite/admin/server.go#L215).
+  The service isn't exposed publicly, the only way to access to it is forwarding the port of the
+  corresponding Kubernetes pod.
+- [Allowed groups](https://github.com/storj/storj/blob/main/satellite/admin/server.go#L231).
+  This requires that the server is accessed through an authorization proxy (e.g.
+  [Oauth2 proxy](https://oauth2-proxy.github.io/)):
+  - AP1: https://ap1.admin.storj.tools
+  - EU1: https://eu1.admin.storj.tools
+  - US1: https://us1.admin.storj.tools
+
+Token secret authentication allows to access al the API endpoints. Allowed groups authentication
+only to:
+- Read user's information.
+- Read/Update user's limits.
+- Freeze/Unfreeze user's account.
+- Unwarn user.
+- Read/Update project's limits.
+
+This current web interface is a visualization of the API. An API isn't an interface for
+non-technical people, as a result, only production owners use the interface or directly the API.
+
+### Goals
+
+We want to have an intuitive Web interface that can be used by people with different backgrounds
+(e.g. Customer support, finance, sales) and becomes it the administration back office of the
+satellite for managing users' accounts.
+
+Back office has a fixed set of access levels. Every access level is associated with a list of
+groups. Users are authorized to specific access level depending to the groups they belong.
+
+Back office keeps a modification history of all performed operations that altered the data (create,
+update,and delete).
+
+Back office can operates without relying in any resource managed by Storj. Any satellite operator
+with or without any relation with Storj will be able to use it.
+
+### Approach / Design
+
+#### Requirements
+
+##### Operations
+
+1. Account:
+  1. View information: ID, email, full name, paid/free tier, suspended?, user agents, Multi-factor
+     authentication enabled/disabled, data placement restriction (geo fence), list of projects,
+     limits (download, storage, number of segments, number of projects allowed).
+  2. Change email address.
+  3. Disable Multi-factor authentication.
+  5. Set limits (download, storage, number of segments, number of projects allowed).
+  6. Suspend "disable upload/download".
```

Comment:
For the purposes of this document, I think this terminology is fine. However, I want to emphasize that we will need to develop some standard universal terminology to distinguish different types of "suspension" for accounts. We currently have multiple ways for a user to become "suspended", and there is different expected behavior for each:
* "billing suspension/freeze" - when a user does not pay invoices, disable upload/download
  * should be unfrozen automatically as soon as invoices are paid
* "prod owner suspension/freeze" - when a user uploads/shares illegal or malicious content, in violation of our terms of service
  * should be permanent, with no way to automatically exit

Right now, we have a lot of confusion about these two situations and there is not clear shared vocabulary across teams, e.g. when Console says "freeze" we are usually referring to "billing freeze", but when support or prod owner says "freeze", they may be talking about a "permanent suspension"

The main takeaway for this back office doc is that we should not go into this thinking that there is only one type of suspend/disable/freeze/whatever we call it - and we should make the differences between these obvious to the operator using the back office.

For the record, I don't have a preference for what terms to use for these different types. But I think we should pick a single word like "freeze" or "suspend" or "disable", and combine it with a second word indicating the type, e.g. "billing freeze" or "permanent freeze" :shrug:

Reply:
Are those different types already available? if yes, please link where I can see them.

> For the record, I don't have a strong preference for what terms to use for these different types. But I think we should pick a single word like "freeze" or "suspend" or "disable", and combine it with a second word indicating the type, e.g. "billing freeze" or "permanent freeze" 🤷

I'd rather use one of those single words with a second word indicating the type too. If I have to vote for one of those words, I vote for "suspend".

Reply:
The types for the "account freezes" table at the time of this comment are [here](https://github.com/storj/storj/blob/c706df52185d99f651963b2e4538b8937310ead3/satellite/console/accountfreezes.go#L70-L78)
Currently, they are just called `Freeze` and `Warning`, and both of them are explicitly related to unpaid invoices (billing suspend).
In order to "permanent suspend" right now, the prod owner must use "set limits" on the user and set to 0, and cannot use "freeze", or the user can exit.
However, soon, I would like to rename `Freeze` and `Warning` to `BillingFreeze` and `BillingWarning`, and add a third type in the section of code I linked to function as "permanent suspend"

TLDR: the type for "permanent suspend" is not available yet, but I think we should add it so that the prod owner does not need to use "set limits" in order to do this

Reply:
Ok, I'll add that new type of suspension

Reply:
Done.
@mobyvb please review that change and the roles that have permissions to perform each one. The changes are in a separated commit, so it should be easy for you to just only check that commit.